rq/queue.py

```diff
@@ -473,6 +475,29 @@ def dequeue_any(cls, queues, timeout, connection=None, job_class=None):
             return job, queue
         return None, None
 
+    @classmethod
+    def redis_mode(cls, connection=None):
+        """Return redis mode
+        :returns: redis mode, cluster(cluster mode) or standalone(single instance)
+        :rtype: str
+        """
+        connection = resolve_connection(connection)
+        info = connection.info()
+        if 'redis_mode' in info:
+            return info['redis_mode']
+        else:
```

Comment:
Is `redis_mode` not always in `connection.info()`?

Reply:
I test `info()` in redis standalon and cluster mode, result is 

pkg | type of `connection ` | type of info  | has key `redis_mode`
------------ | ------------- | ------------- | -------------
[redis-py](https://github.com/andymccurdy/redis-py) | <class 'redis.client.StrictRedis'> | dict | yes
[redis-py-cluster](https://github.com/Grokzen/redis-py-cluster) | <class 'rediscluster.client.StrictRedisCluster'> | dict | each node info has key `redis_mode`

details of command:
```shell
(rediscluster) ➜  ~ python
Python 2.7.10 (default, Oct  6 2017, 22:29:07)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import redis
>>> r = redis.StrictRedis(host='localhost', port=6379, db=0)
>>> r.info()
{'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 0, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 0, 'latest_fork_usec': 0, 'used_cpu_sys': 0.03, 'used_memory_overhead': 1012894, 'maxmemory_human': '0B', 'process_id': 66661, 'repl_backlog_histlen': 0, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.0, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 1036288, 'redis_git_dirty': 0, 'keyspace_hits': 0, 'aof_rewrite_in_progress': 0, 'used_memory_dataset_perc': '3.79%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '991.06K', 'lru_clock': 13463929, 'connected_clients': 1, 'instantaneous_output_kbps': 0.0, 'tcp_port': 6379, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'total_commands_processed': 0, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 0, 'blocked_clients': 0, 'used_memory_dataset': 1954, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '1012.00K', 'run_id': '2b78bbe1f2728ec8d394f6bbc4aa44380bb5f278', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'standalone', 'used_memory_peak_perc': '105.36%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 1014848, 'master_repl_offset': 0, 'active_defrag_key_hits': 0, 'used_cpu_user': 0.02, 'evicted_keys': 0, 'master_replid': 'abdd3b8d7408bd85b096ea4df4acab1a7df32852', 'rdb_last_save_time': 1523413315, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 14, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 0, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 0, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 0, 'used_memory_lua': 37888, 'role': 'master', 'cluster_enabled': 0, 'total_connections_received': 1, 'config_file': '', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 963264, 'used_memory_human': '991.06K', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 0, 'rdb_last_bgsave_time_sec': -1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 0, 'used_memory_peak': 1014848, 'uptime_in_seconds': 54, 'rdb_changes_since_last_save': 0, 'mem_fragmentation_ratio': 1.02}
>>> info = r.info()
>>> type(r)
<class 'redis.client.StrictRedis'>
>>> type(info)
<type 'dict'>
>>> from rediscluster import StrictRedisCluster
>>> startup_nodes = [{"host": "127.0.0.1", "port": "7000"}]
>>> rc = StrictRedisCluster(startup_nodes=startup_nodes, decode_responses=True)
>>> type(rc)
<class 'rediscluster.client.StrictRedisCluster'>
>>> infoc = rc.info()
>>> type(infoc)
<type 'dict'>
>>> infoc
{'127.0.0.1:7000': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 5, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 0, 'latest_fork_usec': 0, 'used_cpu_sys': 31.23, 'used_memory_overhead': 2642652, 'maxmemory_human': '0B', 'process_id': 24516, 'repl_backlog_histlen': 67513, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.02, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 1007616, 'redis_git_dirty': 0, 'keyspace_hits': 30, 'aof_rewrite_in_progress': 0, 'used_memory_dataset_perc': '11.91%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '3.17M', 'lru_clock': 13464665, 'connected_clients': 1, 'instantaneous_output_kbps': 0.0, 'tcp_port': 7000, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'total_commands_processed': 7845, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 155204, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '984.00K', 'run_id': 'a78a907325e0df6505fdc12627f66ad955e0e6fd', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'used_memory_peak_perc': '84.29%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2797856, 'master_repl_offset': 67513, 'active_defrag_key_hits': 0, 'used_cpu_user': 25.48, 'evicted_keys': 0, 'master_replid': 'be6333419807d0aa8507264754b20513fe4ad3b9', 'rdb_last_save_time': 1523283567, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 284707, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 4, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'master', 'slave0': {'ip': '127.0.0.1', 'state': 'online', 'lag': 0, 'port': 7003, 'offset': 67513}, 'cluster_enabled': 1, 'total_connections_received': 777, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.67M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 1, 'rdb_last_bgsave_time_sec': 1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 176834, 'used_memory_peak': 3319296, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 166, 'mem_fragmentation_ratio': 0.36}, '127.0.0.1:7001': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 5, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 1, 'latest_fork_usec': 0, 'used_cpu_sys': 31.01, 'used_memory_overhead': 2609880, 'maxmemory_human': '0B', 'process_id': 24520, 'repl_backlog_histlen': 67301, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.05, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 974848, 'redis_git_dirty': 0, 'keyspace_hits': 44, 'aof_rewrite_in_progress': 0, 'used_memory_dataset_perc': '8.65%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '2.93M', 'lru_clock': 13464665, 'connected_clients': 1, 'instantaneous_output_kbps': 0.0, 'tcp_port': 7001, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'total_commands_processed': 6362, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 105560, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '952.00K', 'run_id': 'f86eb2aa94fcda53f93def38397cd121ac847147', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'used_memory_peak_perc': '88.52%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2715440, 'master_repl_offset': 67301, 'active_defrag_key_hits': 0, 'used_cpu_user': 25.23, 'evicted_keys': 0, 'master_replid': '066c5872fbdcadede13987779cd937e3eb9792e0', 'rdb_last_save_time': 1523283567, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 227630, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 6, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'master', 'slave0': {'ip': '127.0.0.1', 'state': 'online', 'lag': 0, 'port': 7004, 'offset': 67301}, 'cluster_enabled': 1, 'total_connections_received': 267, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.59M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 1, 'rdb_last_bgsave_time_sec': 1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 30598, 'used_memory_peak': 3067744, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 158, 'mem_fragmentation_ratio': 0.36}, '127.0.0.1:7002': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 5, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 1, 'latest_fork_usec': 0, 'used_cpu_sys': 30.66, 'used_memory_overhead': 2609880, 'maxmemory_human': '0B', 'process_id': 24524, 'repl_backlog_histlen': 184911, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.05, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 1011712, 'redis_git_dirty': 0, 'keyspace_hits': 293, 'aof_rewrite_in_progress': 0, 'used_memory_dataset_perc': '8.72%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '2.91M', 'lru_clock': 13464665, 'connected_clients': 1, 'instantaneous_output_kbps': 0.0, 'tcp_port': 7002, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'total_commands_processed': 6964, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 106456, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '988.00K', 'run_id': '75925a2e6ee6662c018921a9e6ad21faf9741e8d', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'used_memory_peak_perc': '89.01%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2716336, 'master_repl_offset': 184911, 'active_defrag_key_hits': 0, 'used_cpu_user': 25.25, 'evicted_keys': 0, 'master_replid': 'aa83798198fcf85c2ca55458d512878eead73401', 'rdb_last_save_time': 1523283567, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 356265, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 31, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'master', 'slave0': {'ip': '127.0.0.1', 'state': 'online', 'lag': 1, 'port': 7005, 'offset': 184911}, 'cluster_enabled': 1, 'total_connections_received': 260, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.59M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 1, 'rdb_last_bgsave_time_sec': 0, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 248239, 'used_memory_peak': 3051760, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 938, 'mem_fragmentation_ratio': 0.37}, '127.0.0.1:7003': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 0, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 0, 'latest_fork_usec': 0, 'used_cpu_sys': 30.96, 'used_memory_overhead': 2609880, 'maxmemory_human': '0B', 'process_id': 24528, 'repl_backlog_histlen': 67513, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.0, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 966656, 'redis_git_dirty': 0, 'keyspace_hits': 0, 'aof_rewrite_in_progress': 0, 'master_sync_in_progress': 0, 'master_host': '127.0.0.1', 'used_memory_dataset_perc': '5.79%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '2.55M', 'lru_clock': 13464665, 'connected_clients': 2, 'slave_repl_offset': 67513, 'instantaneous_output_kbps': 0.02, 'tcp_port': 7003, 'master_last_io_seconds_ago': 7, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'slave_read_only': 1, 'total_commands_processed': 617, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 68488, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '944.00K', 'run_id': '4d3b132bbe6cf32785dce64a1b8394ae59f57609', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'master_link_status': 'up', 'used_memory_peak_perc': '100.00%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2678368, 'master_repl_offset': 67513, 'active_defrag_key_hits': 0, 'used_cpu_user': 23.81, 'evicted_keys': 0, 'master_replid': 'be6333419807d0aa8507264754b20513fe4ad3b9', 'rdb_last_save_time': 1523283556, 'slave_priority': 100, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'master_port': 7000, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 11047, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 0, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'slave', 'cluster_enabled': 1, 'total_connections_received': 18, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.55M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 0, 'rdb_last_bgsave_time_sec': -1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 183452, 'used_memory_peak': 2678368, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 166, 'mem_fragmentation_ratio': 0.36}, '127.0.0.1:7004': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 0, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 0, 'latest_fork_usec': 0, 'used_cpu_sys': 31.09, 'used_memory_overhead': 2609880, 'maxmemory_human': '0B', 'process_id': 24532, 'repl_backlog_histlen': 67301, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.0, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 987136, 'redis_git_dirty': 0, 'keyspace_hits': 2, 'aof_rewrite_in_progress': 0, 'master_sync_in_progress': 0, 'master_host': '127.0.0.1', 'used_memory_dataset_perc': '5.74%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '2.59M', 'lru_clock': 13464665, 'connected_clients': 2, 'slave_repl_offset': 67301, 'instantaneous_output_kbps': 0.05, 'tcp_port': 7004, 'master_last_io_seconds_ago': 7, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'slave_read_only': 1, 'total_commands_processed': 617, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 67960, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '964.00K', 'run_id': '08f4488f6bf8b6d1d818c1a82425eb392a76fba6', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'master_link_status': 'up', 'used_memory_peak_perc': '98.78%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2677840, 'master_repl_offset': 67301, 'active_defrag_key_hits': 0, 'used_cpu_user': 23.95, 'evicted_keys': 0, 'master_replid': '066c5872fbdcadede13987779cd937e3eb9792e0', 'rdb_last_save_time': 1523283556, 'slave_priority': 100, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'master_port': 7001, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 11171, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 0, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'slave', 'cluster_enabled': 1, 'total_connections_received': 19, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.55M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 0, 'rdb_last_bgsave_time_sec': -1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 183415, 'used_memory_peak': 2710992, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 158, 'mem_fragmentation_ratio': 0.37}, '127.0.0.1:7005': {'aof_enabled': 0, 'multiplexing_api': 'kqueue', 'redis_version': '4.0.8', 'sync_partial_ok': 0, 'loading': 0, 'used_memory_lua_human': '37.00K', 'active_defrag_key_misses': 0, 'slave_expires_tracked_keys': 0, 'total_system_memory_human': '8.00G', 'instantaneous_ops_per_sec': 0, 'latest_fork_usec': 0, 'used_cpu_sys': 30.82, 'used_memory_overhead': 2609880, 'maxmemory_human': '0B', 'process_id': 24536, 'repl_backlog_histlen': 184911, 'aof_current_rewrite_time_sec': -1, 'arch_bits': 64, 'aof_last_cow_size': 0, 'pubsub_channels': 0, 'instantaneous_input_kbps': 0.0, 'client_longest_output_list': 0, 'mem_allocator': 'libc', 'used_memory_rss': 1011712, 'redis_git_dirty': 0, 'keyspace_hits': 7, 'aof_rewrite_in_progress': 0, 'master_sync_in_progress': 0, 'master_host': '127.0.0.1', 'used_memory_dataset_perc': '19.20%', 'atomicvar_api': 'atomic-builtin', 'used_memory_peak_human': '2.79M', 'lru_clock': 13464665, 'connected_clients': 2, 'slave_repl_offset': 184911, 'instantaneous_output_kbps': 0.05, 'tcp_port': 7005, 'master_last_io_seconds_ago': 7, 'active_defrag_hits': 0, 'rdb_current_bgsave_time_sec': -1, 'slave_read_only': 1, 'total_commands_processed': 1044, 'aof_last_write_status': 'ok', 'lazyfree_pending_objects': 0, 'pubsub_patterns': 0, 'aof_last_bgrewrite_status': 'ok', 'repl_backlog_active': 1, 'blocked_clients': 0, 'used_memory_dataset': 265032, 'expired_keys': 0, 'total_system_memory': 8589934592, 'repl_backlog_size': 1048576, 'sync_partial_err': 0, 'sync_full': 0, 'os': 'Darwin 17.5.0 x86_64', 'used_memory_rss_human': '988.00K', 'run_id': '2d4a7b8f19fce3ebf90153fabb7d9062e1c8e245', 'redis_build_id': 'd791831611c260ce', 'redis_mode': 'cluster', 'master_link_status': 'up', 'used_memory_peak_perc': '98.35%', 'executable': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/redis-server', 'gcc_version': '4.2.1', 'used_memory': 2874912, 'master_repl_offset': 184911, 'active_defrag_key_hits': 0, 'used_cpu_user': 23.75, 'evicted_keys': 0, 'master_replid': 'aa83798198fcf85c2ca55458d512878eead73401', 'rdb_last_save_time': 1523283556, 'slave_priority': 100, 'hz': 10, 'active_defrag_misses': 0, 'rdb_last_bgsave_status': 'ok', 'aof_last_rewrite_time_sec': -1, 'master_port': 7002, 'client_biggest_input_buf': 0, 'total_net_input_bytes': 127172, 'rdb_bgsave_in_progress': 0, 'second_repl_offset': -1, 'rejected_connections': 0, 'aof_rewrite_scheduled': 0, 'rdb_last_cow_size': 0, 'uptime_in_days': 1, 'redis_git_sha1': 0, 'migrate_cached_sockets': 0, 'keyspace_misses': 3, 'maxmemory': 0, 'used_cpu_user_children': 0.0, 'repl_backlog_first_byte_offset': 1, 'used_memory_lua': 37888, 'role': 'slave', 'cluster_enabled': 1, 'total_connections_received': 19, 'config_file': '/Users/luwenquan/workspace/githubwp/redis-py-cluster/-', 'maxmemory_policy': 'noeviction', 'used_memory_startup': 1494816, 'used_memory_human': '2.74M', 'active_defrag_running': 0, 'master_replid2': 0, 'connected_slaves': 0, 'rdb_last_bgsave_time_sec': -1, 'used_cpu_sys_children': 0.0, 'total_net_output_bytes': 188254, 'used_memory_peak': 2923040, 'uptime_in_seconds': 130549, 'rdb_changes_since_last_save': 938, 'mem_fragmentation_ratio': 0.35}}
```

Reply:
Alright. In that case, may I suggest getting the methods `queue.redis_mode` and `queue.uses_redis_cluster` merged to master via a PR?

Reply:
@selwin Yes, no problem

Reply:
Thanks. We'll get the features merged in gradually, it'll be a long road but the payoff should be awesome ;)